pkg/util/csi/volume_snapshot.go

```diff
@@ -70,6 +72,14 @@ func WaitVolumeSnapshotReady(ctx context.Context, snapshotClient snapshotter.Sna
 		return true, nil
 	})
 
+	if err == wait.ErrWaitTimeout {
+		err = errors.Errorf("volume snapshot is not ready until timeout, errors: %v", errMessage.List())
```

Comment:
which means that we don't do anything if any error happens and wait for a time-out? Is it reasonable?

Reply:
Yes, we don't error out immediately when we see an error is filled because this the errors may be interim. See the description of this error field:
```
field Error *snapshotv1api.VolumeSnapshotError
error is the last observed error during snapshot creation, if any. This field could be helpful to upper level controllers(i.e., application controller) to decide whether they should continue on waiting for the snapshot to be created based on the type of error reported. The snapshot controller will keep retrying when an error occurrs during the snapshot creation. Upon success, this error field will be cleared
```

Reply:
I think it is reasonable until we have a list of common resolvable issues. Most likely, external-snapshotters (csi drivers) will just retry and we hope most errors goes away.
Right now there are no resolvable errors. If errors persists, then we wait for timeout.